.github/workflows/docs.yaml

```diff
@@ -0,0 +1,39 @@
+name: Build and Deploy Documentation
+on:
+  push:
+    branches:
+      - develop
+
+jobs:
+  create_documentation:
+    runs-on: ubuntu-latest
+    name: Deploy documentation
+
+    steps:
+
+    - name: Install ubuntu dependencies
+      run: |
+        echo "Install ubuntu dependencies (if any)"
+
+    - name: Install (upgrade) dependencies
+      run: |
+        pip install --upgrade pip
+        pip install pdoc
+
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Install uwtools
+      run: |
+        pip install .
+
+    - name: Generate documentation
+      run: |
+        pdoc --docformat "google" -o docs uwtools
+        ls -lR docs/
+
+    - name: Deploy documentation
+      uses: JamesIves/github-pages-deploy-action@4.1.4
```

Comment:
The concern is noted.
If there is a way to replicate this action by pure Github commands, then by all means we can replace this with our home-grown stuff. 
IMO, it is not worth the time and effort.
This action has 2600+ stars and more forks than the ufs-weather-model.  
if we would rather have this forked and us use the fork, then that would work as well.  But then we maintain the fork.

Reply:
Re. ReadTheDocs, I am open to it.
When we have that page, we can deploy there and turn off here.
The documentation will be the same, be it here or there.  Its just a matter of who hosts it where.

Reply:
I'm curious if Julie knows more about a trusted Action to use here.

Reply:
What the advantage of viewing the docs on ReadTheDocs website as opposed to 
https://ufs-community.github.io/workflow-tools

Reply:
No idea. That's why I would like Julie to weigh in.

Reply:
We can initiate a discussion and an amending PR following it.
Like I said, this is a basic PR to get things going.

Reply:
One advantage of viewing the docs on ReadTheDocs as opposed using to the github.io is the version selector and other options (ability to create a pdf, automatic links to the GitHub repo, etc.) provided in the lower left corner.  See attached screenshot.  For the METplus components, the version selector was essential for our documentation.

Another advantage is the automatic building and rendering of feature branches (set up with automated rules by the admin - see second screenshot below).  The develop does not have to build the documentation themselves and can easily view the documentation for their branch on ReadTheDocs, which makes our pull requests easier as the developer can just link to the rendered documentation in their PR and the reviewer can view it that way.

ReadTheDocs also collects Version Analytics and Search Analytics and has automated features like emailing folks if a documentation build fails.

Overall, our switch to ReadTheDocs entailed less maintenance for us.  I'm having a hard time remembering specifics, but our whole team has felt that the switch to ReadTheDocs was worth it.

Screenshot:
<img width="747" alt="Screen Shot 2022-02-02 at 4 39 04 PM" src="https://user-images.githubusercontent.com/11999822/152256308-db4c9012-1cbe-4cdd-bf1f-712140eb6866.png">

Screenshot:
<img width="1173" alt="Screen Shot 2022-02-02 at 4 47 51 PM" src="https://user-images.githubusercontent.com/11999822/152256938-d0f007f5-cd4a-450e-aa5d-38f823659568.png">

Reply:
Regarding:

> I am concerned about the use of a GitHub action supplied by an individual. Is there an option to use an action from a more trusted 3rd-party?

I'm not sure I fully understand, but I passed this information along to our GitHub Actions lead and he said the following, which I hope will be useful:

> There are GitHub Actions provided on the GitHub Marketplace that you can use in workflows for common tasks, such as checking out the code from GitHub.
Here is an example: https://github.com/marketplace/actions/checkout
You can see on the right side of that page that this was created by a "Verified creator".
There are some actions that are provided on the marketplace that were created by 3rd party contributors.  Here is an example that John HG found: https://github.com/marketplace/actions/cancel-workflow-action
You can see on the right side of this one that it says:
Cancel Workflow Action is not certified by GitHub. It is provided by a third-party and is governed by separate terms of service, privacy policy, and support documentation.

> For the METplus repositories, we currently do not use any actions that are not verified i.e. created by GitHub

> The source code for these actions are publicly available on GitHub, so they can be reviewed for malicious behavior. It is advised to use specific versions of these actions, i.e. @4.1.4., to ensure that the tagged version of the code is used.

> This is the action that was mentioned on marketplace: https://github.com/marketplace/actions/deploy-to-github-pages.

> We don't use GitHub pages any more, so that isn't relevant to us. We use ReadTheDocs which doesn't require any GHA automation to handle. I would think there are other actions available to handle this step. If not, they can always implement the logic themselves in a script.

Reply:
@jprestop 
Thank you for your input.
I understand the benefits of RTD.  gh-pages by default does not do versioned documentation, but there are github actions available in the marketplace that do this. Having said that, we would like to establish the RTD pipeline before making a release of this package.
Would it be possible for someone from your team to guide us setup RTD and the (automatic) connections from Github to RTD?  I am not fluent in those requirements and process.

Reply:
Also, just to note, we are a long long way from putting, what I see these as "finishing touches".  
As I have described in the PR, this is simply a placeholder to get us started.

Reply:
@aerorahul 
> Would it be possible for someone from your team to guide us setup RTD and the (automatic) connections from Github to RTD? 

I was just talking with Tara about this today.  She is willing to donate up to two days of my time to either set this up for the UFS workflow team or provide guidance to set it up.  I don't remember the details of setting it up, but I remember it was fairly easy and didn't take too long.  It shouldn't take too long to refresh my memory.  I'm happy to help out with this in whatever way you feel is best for the UFS workflow team.

Reply:
That would be wonderful. 
Thank you so much. 
I will amend this PR.

Reply:
Potentially dumb question from someone who missed that this conversation was happening and is trying to get caught up - does this need to go into this PR, or can we split this out and tackle it separately when @jprestop has time on her calendar to do it?

Reply:
I believe I removed the deployment.  It will be added in a later PR.